docs/griptape-framework/structures/task-memory.md

```diff
@@ -1,89 +1,320 @@
+[04/26/24 14:44:41] INFO     Subtask 51f637e66a7d415488b4912834475bee
+                             Thought: First, I need to scrape the content of the webpage. Then, I will summarize the content and finally save the summary to a file.
+                             Actions: [{"name": "WebScraper", "path": "get_content", "input": {"values": {"url": "https://simple.wikipedia.org/wiki/San_Francisco"}},
+                             "tag": "scrape_webpage"}]
+[04/26/24 14:44:46] INFO     Subtask 51f637e66a7d415488b4912834475bee
+                             Response: Output of "WebScraper.get_content" was stored in memory with memory_name "TaskMemory" and artifact_namespace
+                             "9a43acd36a24440cb436c0fef189c534"
+[04/26/24 14:44:51] INFO     Subtask 269c537132b3461386c5032cea719805
+                             Thought: Now that the webpage content is stored in memory, I need to summarize it.
+                             Actions: [{"tag": "summarize_content", "name": "TaskMemoryClient", "path": "summarize", "input": {"values": {"memory_name": "TaskMemory",
+                             "artifact_namespace": "9a43acd36a24440cb436c0fef189c534"}}}]
+[04/26/24 14:45:57] INFO     Subtask 269c537132b3461386c5032cea719805
+                             Response: Output of "TaskMemoryClient.summarize" was stored in memory with memory_name "TaskMemory" and artifact_namespace
+                             "fe04d7543ac64aecab6b00f5417f5bfb"
+[04/26/24 14:46:02] INFO     Subtask 2618fc23194947018ce7d3facb33e5b9
+                             Thought: Now that the summary is stored in memory, I need to save it to a file.
+                             Actions: [{"tag": "save_summary", "name": "FileManager", "path": "save_memory_artifacts_to_disk", "input": {"values": {"dir_name":
+                             "summaries", "file_name": "san_francisco_summary.txt", "memory_name": "TaskMemory", "artifact_namespace":
+                             "fe04d7543ac64aecab6b00f5417f5bfb"}}}]
+                    INFO     Subtask 2618fc23194947018ce7d3facb33e5b9
+                             Response: Successfully saved memory artifacts to disk
+[04/26/24 14:46:04] INFO     ToolkitTask 1ed8480fa2dc42d7ac56165491869efc
+                             Output: The summary of the webpage has been successfully saved to the file "san_francisco_summary.txt" in the "summaries" directory.
+
 ```
+
+Note that in this example GPT-4 _never_ saw the contents of the page, only that it was stored in Task Memory.
+
+!!! info
+    In this instance of using `FileManager` to save a file to disk, it does not matter what the value of `off_prompt` is set to. This Tool Activity returns an `InfoArtifact` which Griptape will not store in Task Memory. Griptape will only store `TextArtifact`'s, `BlobArtifact`'s, and `ListArtifact`'s in Task Memory.
```

Comment:
This may be good to call out with a section header like `## When using FileManager` or `## Notes on InfoArtifacts`.